docs/tools/testing/icetea.md

```diff
@@ -0,0 +1,192 @@
+        expected_retcode = <int>    # expected return code (default=0)
+        timeout=<int>              # timeout, if no retcode receive
+```
+
+### Documentation
+
+You can build HTML documentation for Icetea using sphinx. The source for the documentation is located in [doc-source](https://github.com/ARMmbed/icetea/tree/master/doc-source). For installation of sphinx, please see the [installation instructions](#installation).
+
+A build script for the documentation is included in `build_docs.py`. It's a Python script that also generates autodoc API documentation. Run the script with:
+
+`python build_docs.py`
+
+### Examples
+
+**Note:** The following examples use the [`dummyDut`](https://github.com/ARMmbed/icetea/blob/master/test/dut/dummyDut.c) application. It works in unix systems. To run the following examples, please compile `dummyDut` using Make:
+
+`> make -C test/dut`.
+
+To print available parameters:
+
+`> icetea --help`
+
+To run a single test case with the process DUT:
+
+`> icetea --tc test_cmdline --tcdir examples --type process --bin ./test/dut/dummyDut`
+
+To run all existing test cases from the `examples` folder:
+
+`> icetea --tc all --tcdir examples --type process --bin ./test/dut/dummyDut`
+
+To debug DUT locally with [GDB](https://www.gnu.org/software/gdb/):
+
+**Note:** You have to install [GDB](https://www.gnu.org/software/gdb/) first (`apt-get install gdb`).
+
+```
+> icetea --tc test_cmdline --tcdir examples --type process --gdb 1 --bin ./test/dut/dummyDut
+> sudo gdb ./CliNode 3460
+```
+
+To debug DUT remotely with the GDB server:
+
+```
+> icetea --tc test_cmdline --tcdir examples --type process --gdbs 1 --bin  ./test/dut/dummyDut
+> gdb  ./test/dut/dummyDut --eval-command="target remote localhost:2345"
+```
+
+To analyze memory leaks with valgrind:
+
+**Note:** You have to install [valgrind](http://valgrind.org) first (`apt-get install valgrind`):
+
+```
+> icetea --tc test_cmdline --tcdir examples --type process --valgrind --valgrind_tool memcheck --bin  ./test/dut/dummyDut
+```
+
+### Running unit tests with *Icetea*
+
+To build a test application for DUT and execute the test:
+
+```
+> make
+> coverage run -m unittest discover -s test
+```
+
+To generate a coverage report (excluding plugins):
+
+```
+> coverage html --include "icetea_lib/*" --omit "icetea_lib/Plugin/plugins/*"
+```
+
+To run unit tests for plugins that ship with Icetea:
+
+```
+> coverage run -m unittest discover -s icetea_lib/Plugin/plugins/plugin_tests
+```
+
+To generate a coverage reports for plugin unit tests, run:
+
+```
+> coverage html --include "icetea_lib/Plugin/plugins/*" --omit "icetea_lib/Plugin/plugins/plugin_tests/*"
+```
+
+#### Dependencies
+
+Unit tests depend on mock, coverage and netifaces.
+
+```
+> pip install mock
+> pip install netifaces
+> pip install coverage
+```
+
+### License
+
+See the [license](https://github.com/ARMmbed/icetea/blob/master/LICENSE) agreement.
```

Comment:
Does this have a different license than the rest of our tools? Can we add its information to the contributing section with the rest of our licensing content?